### PR TITLE
feat: summary of work survey for dashboard

### PR DESCRIPTION
**Pull request by datphamcode295**
#### What's this PR do?

- [x] Summary of work survey for dashboard

#### What are the relevant Git tickets?
https://www.notion.so/dwarves/GET-API-to-fetch-summary-of-work-survey-based-on-projects-729af5e0ec8d46198fbaacefb60f3a39

#### Screenshots (if appropriate)
image

#### Request
```
curl --location --request GET 'http://localhost:8200/api/v1/dashboards/work-surveys?projectID=8dc3be2e-19a4-4942-8a79-56db391a0b15' \
--header 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2OTkzMjExNDIsImlkIjoiMjY1NTgzMmUtZjAwOS00YjczLWE1MzUtNjRjM2EyMmU1NThmIiwiYXZhdGFyIjoiaHR0cHM6Ly9zMy1hcC1zb3V0aGVhc3QtMS5hbWF6b25hd3MuY29tL2ZvcnRyZXNzLWltYWdlcy81MTUzNTc0Njk1NjYzOTU1OTQ0LnBuZyIsImVtYWlsIjoidGhhbmhAZC5mb3VuZGF0aW9uIiwicGVybWlzc2lvbnMiOlsiZW1wbG95ZWVzLnJlYWQiXSwidXNlcl9pbmZvIjpudWxsfQ.GENGPEucSUrILN6tHDKxLMtj0M0REVMUPC7-XhDMpGM' \
--data-raw ''
```

#### Response
```
{
    "data": {
        "project": {
            "id": "8dc3be2e-19a4-4942-8a79-56db391a0b15",
            "name": "Fortress",
            "type": "dwarves",
            "status": "active",
            "code": "fortress"
        },
        "workSurveys": [
            {
                "endDate": "28/02",
                "workload": 0,
                "deadline": 3,
                "learning": 1,
                "trend": null
            },
            {
                "endDate": "21/02",
                "workload": 0,
                "deadline": 0,
                "learning": 0,
                "trend": {
                    "workload": 0,
                    "deadline": 0,
                    "learning": 0
                }
            },
            {
                "endDate": "07/02",
                "workload": 0,
                "deadline": 0,
                "learning": 0,
                "trend": {
                    "workload": 0,
                    "deadline": 0,
                    "learning": 0
                }
            },
            {
                "endDate": "23/01",
                "workload": 0,
                "deadline": 0,
                "learning": 0,
                "trend": {
                    "workload": 0,
                    "deadline": 0,
                    "learning": 0
                }
            },
            {
                "endDate": "09/01",
                "workload": 0,
                "deadline": 0,
                "learning": 0,
                "trend": {
                    "workload": 0,
                    "deadline": 0,
                    "learning": 0
                }
            },
            {
                "endDate": "01/01",
                "workload": 0,
                "deadline": 0,
                "learning": 0,
                "trend": {
                    "workload": 0,
                    "deadline": 0,
                    "learning": 0
                }
            }
        ]
    }
}
```

### CODE REVIEW

File: pkg/store/dashboard/dashboard.go [hunk 1]
```diff
@@ -1,7 +1,101 @@
...
 type store struct{}
 
 func New() IStore {
 	return &store{}
 }
+
+func (s *store) GetWorkSurveysByProjectID(db *gorm.DB, projectID string) ([]*model.WorkSurvey, error) {
+	var rs []*model.WorkSurvey
+
+	query := `
```
Comment (nguyennh4522) [inline pkg/store/dashboard/dashboard.go:17]: format it

File: pkg/store/dashboard/dashboard.go [hunk 2]
```diff
@@ -1,7 +1,101 @@
...
+			limit 6
+	`
+
+	return rs, db.Debug().Raw(query, projectID).Scan(&rs).Error
+}
+
+func (s *store) GetAllWorkSurveys(db *gorm.DB) ([]*model.WorkSurvey, error) {
+	var rs []*model.WorkSurvey
+
+	query := `
```
Comment (nguyennh4522) [inline pkg/store/dashboard/dashboard.go:60]: format it

Comment (datphamcode295) [reply pkg/store/dashboard/dashboard.go:56]: updated

File: migrations/seed/permissions.sql [hunk 3]
```diff
@@ -39,4 +39,8 @@ INSERT INTO public.permissions (id, deleted_at, created_at, updated_at, name, co
 ('f632f18c-98da-4c1a-b2c6-9ccd0c7c5a39', NULL, '2022-11-11 18:34:14.743263', '2022-11-11 18:34:14.743263', 'Employee Roles Create', 'employeeRoles.create'),
 ('5c5919b5-e74b-4854-bdf6-b91396a8317c', NULL, '2022-11-11 18:34:14.743263', '2022-11-11 18:34:14.743263', 'Employee Roles Read', 'employeeRoles.read'),
 ('68fe5bcf-38ae-446a-8cbb-8f88b5c6eb44', NULL, '2022-11-11 18:34:14.743263', '2022-11-11 18:34:14.743263', 'Employee Roles Edit', 'employeeRoles.edit'),
-('98744d43-f14b-4235-8dfc-5265668daa21', NULL, '2022-11-11 18:34:14.743263', '2022-11-11 18:34:14.743263', 'Employee Roles Delete', 'employeeRoles.delete');
+('98744d43-f14b-4235-8dfc-5265668daa21', NULL, '2022-11-11 18:34:14.743263', '2022-11-11 18:34:14.743263', 'Employee Roles Delete', 'employeeRoles.delete'),
+('4be48741-f8b2-4562-964e-65c081fec7a9', NULL, '2022-11-11 18:34:14.743263', '2022-11-11 18:34:14.743263', 'DashBoard Create', 'dashboards.create'),
```
Comment (nguyennh4522) [inline migrations/seed/permissions.sql:43]:
```suggestion
('4be48741-f8b2-4562-964e-65c081fec7a9', NULL, '2022-11-11 18:34:14.743263', '2022-11-11 18:34:14.743263', 'Dashboard Create', 'dashboards.create'),
```

File: pkg/store/dashboard/dashboard.go (hunk 2 above)
Comment (nguyennh4522) [reply pkg/store/dashboard/dashboard.go:56]: remove `(` `)` unnecessary, put the work of the same level in the same column, and uppercase the words like `CASE`, `WHEN`,...

**Review by thanhpn — APPROVED** [review]

**Pull request merged**